### PR TITLE
Update tools, add nrfutil and checksum verification

### PR DESCRIPTION
**Pull request by whati001**
Update command line tool version, added checksum validation and nrfutil.

### CODE REVIEW

File: nrfconnect-toolchain/Dockerfile [hunk 1]
```diff
@@ -30,8 +30,10 @@
 
 FROM ubuntu:21.04
 
-ARG TOOLCHAIN_URL="https://developer.arm.com/-/media/Files/downloads/gnu-rm/9-2019q4/gcc-arm-none-eabi-9-2019-q4-major-x86_64-linux.tar.bz2"
-ARG NRF_TOOLS_URL="https://www.nordicsemi.com/-/media/Software-and-other-downloads/Desktop-software/nRF-command-line-tools/sw/Versions-10-x-x/10-14-0/nRF-Command-Line-Tools_10_14_0_Linux64.zip"
+ARG TOOLCHAIN_MD5="fe0029de4f4ec43cf7008944e34ff8cc"
+ARG TOOLCHAIN_URL="https://armkeil.blob.core.windows.net/developer/Files/downloads/gnu-rm/9-2019q4/gcc-arm-none-eabi-9-2019-q4-major-x86_64-linux.tar.bz2"
+ARG NRF_TOOLS_SHA256="05982D0E104A8723D2EA1BFD5AA12493C87164DE2D3A1D62C8B2D21BD40CCB88"
+ARG NRF_TOOLS_URL="https://nsscprodmedia.blob.core.windows.net/prod/software-and-other-downloads/desktop-software/nrf-command-line-tools/sw/versions-10-x-x/10-15-1/nrf-command-line-tools-10.15.1_linux-amd64.zip"
```
Comment (Damian-Nordic) [inline nrfconnect-toolchain/Dockerfile:36]: Could you use a more official address: `https://www.nordicsemi.com/-/media/Software-and-other-downloads/Desktop-software/nRF-command-line-tools/sw/Versions-10-x-x/10-15-1/nrf-command-line-tools-10.15.1_Linux-amd64.zip`? I can see you've added `-L` to the curl invocation, so it should redirect automatically, right?

File: nrfconnect-toolchain/Dockerfile [hunk 2]
```diff
@@ -30,8 +30,10 @@
 
 FROM ubuntu:21.04
 
-ARG TOOLCHAIN_URL="https://developer.arm.com/-/media/Files/downloads/gnu-rm/9-2019q4/gcc-arm-none-eabi-9-2019-q4-major-x86_64-linux.tar.bz2"
```
Comment (Damian-Nordic) [inline nrfconnect-toolchain/Dockerfile:33]: Is this change necessary? Isn't adding `-L` to `curl` sufficient?

**Review by Damian-Nordic — COMMENTED** [review]
Thanks for the contribution :) Have two questions/comments.

File: nrfconnect-toolchain/Dockerfile (hunk 2 above)
Comment (whati001) [reply nrfconnect-toolchain/Dockerfile:33]: yes -L is sufficient. I have removed -L first, then updated the URL to the redirect one and afterwards added -L because we check the hash now.
Should I use the old "more official" one again?

File: nrfconnect-toolchain/Dockerfile (hunk 1 above)
Comment (whati001) [reply nrfconnect-toolchain/Dockerfile:36]: same answer as below

File: nrfconnect-toolchain/Dockerfile (hunk 2 above)
Comment (Damian-Nordic) [reply nrfconnect-toolchain/Dockerfile:33]: Yes, please. I think the redirect URL is more likely to change, so let's use the old one.

File: nrfconnect-toolchain/Dockerfile (hunk 1 above)
Comment (whati001) [reply nrfconnect-toolchain/Dockerfile]: Reversed the change for TOOLCHAIN_URL and used the more official link for the cmd tools.
Taken from [www.nordicsemi.com]( https://www.nordicsemi.com/Products/Development-tools/nRF-Command-Line-Tools/Download?lang=en#infotabs)

File: nrfconnect-chip/Dockerfile [hunk 3]
```diff
@@ -60,6 +60,10 @@ RUN set -x \
     && python3 -c "from zipfile import *; ZipFile('/tmp/gn.zip').extract('gn', '/usr/bin')" \
     && chmod +x /usr/bin/gn \
     #
+    # Install nrfutil
+    #
+    && python3 -m pip install --no-cache-dir nrfutil \
```
Comment (whati001) [inline nrfconnect-chip/Dockerfile:65]: install nrfutil, which is nice for flashing non dk devices like dongels

File: nrfconnect-toolchain/Dockerfile [hunk 4]
```diff
@@ -40,20 +42,25 @@ RUN set -x \
     #
     && DEBIAN_FRONTEND=noninteractive apt-get update \
     && DEBIAN_FRONTEND=noninteractive apt-get -yq install --no-install-recommends bzip2 unzip curl ninja-build python3-pip git device-tree-compiler libusb-1.0-0 \
-    && DEBIAN_FRONTEND=noninteractive apt-get -yq install --no-install-recommends gcc libpython3-dev \
+    && DEBIAN_FRONTEND=noninteractive apt-get -yq install --no-install-recommends gcc libpython3-dev libsm6 \
```
Comment (whati001) [inline nrfconnect-toolchain/Dockerfile:45]: libsm6 is needed for the newer command line tools

**Review by Damian-Nordic — APPROVED** [review]

**Comment by Damian-Nordic** [comment]
@whati001 FYI, a new image 0.1.12 has been uploaded to the dockerhub.
